lib/collection/src/operations/types.rs

```diff
@@ -1258,12 +1258,12 @@ impl Record {
         }
     }
 
-    pub fn get_vector_by_name(&self, name: &str) -> Option<VectorRef> {
+    pub fn get_vector_by_name(&self, name: &str) -> Option<Vector> {
```

Comment:
what about `&Vector`?

Reply:
`self.vector` here has a type `Option<VectorStruct>`. Let's look into `VectorStruct` definition:
```
pub enum VectorStruct {
    Single(DenseVector),
    Multi(HashMap<String, Vector>),
}
```
In case of single-vector without name, we cannot convert `&DenseVector` into `&Vector` type. That's why I have taken ownership here because we can convert `DenseVector` into `Vector`

Reply:
Is there really no other way to make it happen?

I can imagine these clones might be coming somewhat expensive if we do it everywhere.